### PR TITLE
Improve sql-injection protection for sql_sort

### PR DESCRIPTION
**Pull request by PeterChaplin**
- Tests run sucessfully
- Tested a report with sql_sort on mla_erp DEV
- Audited mla-erp code for unusual usage of sql_sort, nothing found.

### CODE REVIEW

File: test/sql.test [hunk 1]
```diff
@@ -41,17 +41,27 @@ namespace eval ::qcode::test {
     test sql_sort-1.0 {sql_sort} -setup {
     } -body {
         sql_sort name email user_code 
-    } -cleanup {} -result {name,email,user_code}
+    } -cleanup {} -result {"name","email","user_code"}
```
Comment (daniel-qcode) [inline test/sql.test]: what happens when you call something like `qc::sql_sort column_a DESC NULLS LAST`?
Is this pattern ever used for MLA?

Comment (daniel-qcode) [inline test/sql.test]: Guess this doesn't apply cause you would just use the `-nulls` option for this.
Can you just quickly check MLA to make sure though?

File: tcl/sql.tcl [hunk 2]
```diff
@@ -129,7 +130,7 @@ proc qc::sql_sort { args } {
         }
         upset 1 offset $offset
 
-        return "$sql limit $limit offset $offset"
+        return "$sql limit [qc::db_quote $limit] offset [qc::db_quote $offset]"
```
Comment (daniel-qcode) [inline tcl/sql.tcl]: should we set the type for integer when calling db_quote for limit and offset?

File: test/sql.test [hunk 3]
```diff
@@ -41,17 +41,27 @@ namespace eval ::qcode::test {
...
 
     test sql_sort-1.1 {sql_sort with order} -setup {
     } -body {
         sql_sort name DESC email user_code 
-    } -cleanup {} -result {name DESC NULLS LAST,email,user_code}
+    } -cleanup {} -result {"name" DESC NULLS LAST,"email","user_code"}
 
     test sql_sort-1.2 {sql_sort with paging} -setup {
     } -body {
```
Comment (daniel-qcode) [inline test/sql.test]: can you add a test case for the `-nulls` option.

File: test/sql.test (hunk 1 above)
Comment (PeterChaplin) [inline test/sql.test]: I'm pretty sure that would have been an error under the existing code, this code shouldn't affect that.

The correct usage would be `qc::sql_sort -nulls last -- column_a DESC`, which should continue to work just fine (see test case 1.1)

Comment (PeterChaplin) [inline test/sql.test]: The existing code would generate:
`column_a DESC, NULLS, LAST` which would be a sql error (note the commas)

File: tcl/sql.tcl (hunk 2 above)
Comment (PeterChaplin) [inline tcl/sql.tcl]: I don't think it's necessary.

Comment (daniel-qcode) [inline tcl/sql.tcl]: ok

File: test/sql.test (hunk 3 above)
Comment (PeterChaplin) [inline test/sql.test]: Added

**Pull request merged**